src/commands/synthetics/utils/internal.ts

```diff
@@ -253,11 +255,12 @@ export const validateAndParseOverrides = (overrides: string[] | undefined): Accu
           }
           break
 
-        // Convert to {[key: string]: string}
+        // Convert to StringObject
         case 'headers':
+        case 'variables':
           if (subKey) {
-            acc['headers'] = acc['headers'] ?? {}
-            acc['headers'][subKey] = value
+            const stringObject: StringObject = acc[key] = acc[key] ?? {} // acc[key] could be undefined, because of potential js Proxy
```

Comment:
> because of potential js Proxy

Did this happen? Also, I'm not sure what you mean - and is it the only possibility of `acc[key]` to be `undefined`?

Reply:
No but if you don't do that, ts still thinks `acc[key]` can be undefined and won't compile...

Reply:
Which is good - I don't think you need the comment then, I was confused about it 😉